pandas/core/frame.py

```diff
@@ -4478,8 +4482,26 @@ def query(self, expr: str, *, inplace: bool = False, **kwargs) -> DataFrame | No
             raise ValueError(msg)
         kwargs["level"] = kwargs.pop("level", 0) + 1
         kwargs["target"] = None
-        res = self.eval(expr, **kwargs)
+        
+        
+        try:
+            res = self.eval(expr, **kwargs)
```

Comment:
The bug in `eval` should be addressed directly instead of falling back like this